client.go

```diff
@@ -114,179 +81,46 @@ type httpsClientConfig struct {
-	if err != nil {
-		log.Error("Can not fetch update image: ", err)
-		return nil, -1, err
-	}
-
-	log.Debugf("Received fetch update response %v+", r)
-
-	if r.StatusCode != http.StatusOK {
-		log.Errorf("Error fetching shcheduled update info: code (%d)", r.StatusCode)
-		return nil, -1, errors.New("Error receiving scheduled update information.")
-	}
-
-	if r.ContentLength < 0 {
-		return nil, -1, errors.New("Will not continue with unknown image size.")
-	} else if r.ContentLength < c.minImageSize {
-		log.Errorf("Image smaller than expected. Expected: %d, received: %d", c.minImageSize, r.ContentLength)
-		return nil, -1, errors.New("Image size is smaller than expected. Aborting.")
-	}
-
-	return r.Body, r.ContentLength, nil
-}
-
-func (client *httpClient) makeAndSendRequest(method, url string) (*http.Response, error) {
-
-	res, err := http.NewRequest(method, url, nil)
-	if err != nil {
-		return nil, err
-	}
-
-	log.Debug("Sending HTTP [", method, "] request: ", url)
-	return client.HTTPClient.Do(res)
-}
-
-// possible API responses received for update request
-const (
-	updateResponseHaveUpdate = 200
-	updateResponseNoUpdates  = 204
-	updateResponseError      = 404
-)
-
-// have update for the client
-type UpdateResponse struct {
-	Image struct {
-		URI      string
-		Checksum string
-		YoctoID  string `json:"yocto_id"`
-		ID       string
+		return nil, errorLoadingClientCertificate
 	}
-	ID string
+	return &clientCert, nil
 }
 
-func validateGetUpdate(update UpdateResponse) error {
-	// check if we have JSON data correctky decoded
-	if update.ID != "" && update.Image.ID != "" &&
-		update.Image.URI != "" && update.Image.YoctoID != "" {
-		log.Info("Correct request for getting image from: " + update.Image.URI)
-		return nil
-	}
-	return errors.New("Missing parameters in encoded JSON response")
-}
-
-func processUpdateResponse(response *http.Response) (interface{}, error) {
-	log.Debug("Received response:", response.Status)
-
-	respBody, err := ioutil.ReadAll(response.Body)
-	if err != nil {
-		return nil, err
-	}
-
-	switch response.StatusCode {
-	case updateResponseHaveUpdate:
-		log.Debug("Have update available")
-
-		data := UpdateResponse{}
-		if err := json.Unmarshal(respBody, &data); err != nil {
-			switch err.(type) {
-			case *json.SyntaxError:
-				return nil, errors.New("Error parsing data syntax")
-			}
-			return nil, errors.New("Error parsing data: " + err.Error())
-		}
-		if err := validateGetUpdate(data); err != nil {
-			return nil, err
-		}
-		return data, nil
-
-	case updateResponseNoUpdates:
-		log.Debug("No update available")
-		return nil, nil
-
-	case updateResponseError:
-		return nil, errors.New("Client not authorized to get update schedule.")
-
-	default:
-		return nil, errors.New("Invalid response received from server")
+func buildURL(server string) string {
+	if strings.HasPrefix(server, "https://") || strings.HasPrefix(server, "http://") {
+		return server
 	}
+	return "https://" + server
```

Comment:
Just a note here as this is still bit messy without server supporting HTTPS and client not having authentication implemented. I am not sure if returning `https` by default is the correct way. Maybe we can see and modify later once the code will be more mature.